### ISSUE TITLE
Set Title dialog box appears when double-clicking in "tui" apps

### ISSUE THREAD

**Issue by mjd-tech** [mate-desktop/mate-terminal#444]
#### Expected behaviour
Dialog should only appear when double-clicking a tab

#### Actual behaviour
When running "tui" apps such as vim, neovim, htop, midnight commander, etc.
the dialog appears when double-clicking in the terminal window.

#### Steps to reproduce the behaviour
Launch mate-terminal. run vim. Double-click anywhere inside the window.

#### MATE general version
1.26.1

#### Package version
1.27.1

#### Linux Distribution
Manjaro

#### Link to bugreport of your Distribution (requirement)
I attempted to do this but failed

**Comment by lukefromdc** [comment]
I wasn't able to duplicate this with visudo (based on vim) or with nano. I don't have vim itself installed as I prefer nano for such tasks

**Comment by mjd-tech** [comment]
You're right. 
Not a problem with nano or less (the pager). 
I've noticed it with: htop, mc (midnight commander), gdu, vim, neovim, dialog. 
Maybe these use a different ncurses library or something.
When I use Tilix instead of Mate-terminal, everything works as expected.
Mate-terminal was working fine until the "double-click tab to set title" feature was added.

**Comment by lukefromdc** [comment]
I don't have anything installed by default that triggers this. Presumably some of the other devs here do

**Comment by mjd-tech** [comment]
Thanks for looking at this.

Just to make sure this isn't a Manjaro/Arch problem, I did the following:
- Installed Debian Testing with Mate Desktop, in a VM.
- built mate-terminal 1.27.1 according to instructions. 

Same issues as on Manjaro, so it's not a distro issue.

I took a look at xfce4-terminal's terminal-window.c 
at line 1196, it looks like they are checking to make sure you actually double clicked on a tab.

Whereas in mate-terminal's terminal-window.c, line 2994 it just acts on a double click event.

Perhaps if we add xfce's check, it fixes the problem.
Unfortunately I don't know how to do this.

Here's the snippet from xfce4-terminal
```
        /* check if the user double-clicked on the label */
          label = gtk_notebook_get_tab_label (notebook, GTK_WIDGET (window->priv->active));
          if (terminal_window_notebook_event_in_allocation (x, y, label))
            {
              terminal_window_action_set_title (window);
              return TRUE;
            }
        }
```

**Comment by lukefromdc** [comment]
Looks like the general algorithm here is to ensure the click was in the tab space before opening the dialog

**Comment by lukefromdc** [comment]
Installed htop, issue confirmed here

**Comment by lukefromdc** [comment]
Try
https://github.com/mate-desktop/mate-terminal/pull/446

**Comment by lukefromdc** [comment]
We have a better and simpler fix in https://github.com/mate-desktop/mate-terminal/pull/446 now, thanks to https://github.com/cwendling

**Comment by lukefromdc** [comment]
https://github.com/mate-desktop/mate-terminal/pull/446 is merged, so closing as the fix will automatically be in the next 1.27 point release
